### ISSUE TITLE
Update check fails with error in log (silently to users) if update already downloaded/pending

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#15028]
### Steps to reproduce:

1. Go to NVDA -> Help -> Check for update. This should be done when there is an update available.
2. Choose to download update.
3. Choose to install later.
4. Restart NVDA
5. Go to NVDA -> Help -> Check for update

### Actual behavior:

* "Checking for update" is spoken but interrupted.
*  When running in an alpha, there is an error sound, and no speech thereafter.
* After the aborted "checking for update", controls in the current app do not work correctly, and one must do an alt+tab twice, or similar app context reset, in order to restore functionality.
* The following appears in the log:

```
IO - speech.speech.speak (04:36:52.816) - MainThread (5624):
Speaking ['Checking for Update', 'dialog', 'Checking for update', CancellableSpeech (still valid)]
DEBUG - gui.contextHelp.bindHelpEvent (04:36:53.035) - MainThread (5624):
Did context help binding for UpdateResultDialog
ERROR - unhandled exception (04:36:53.035) - MainThread (5624):
Traceback (most recent call last):
  File "wx\core.pyc", line 3407, in <lambda>
  File "updateCheck.pyc", line 381, in __init__
TypeError: bad operand type for unary +: 'str'
DEBUGWARNING - IAccessibleHandler.internalWinEventHandler._shouldGetEvents (04:36:53.165) - MainThread (5624):
Foreground took too long to change. Foreground still 4130496 (wxWindowNR). Should be 65552 (#32769)
```

### Expected behavior:

Either the update check should do the same as the "install pending update" does, once an update has been downloaded; or it should delete the pending update and check as normal.

Either way, erroring out or not alerting the user to what is happening, is a bug.

### NVDA logs, crash dumps and other attachments:

[nvda_updateCheckFailed.txt](https://github.com/nvaccess/nvda/files/11786212/nvda_updateCheckFailed.txt)

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

alpha-28430,3920fa58

#### Windows version:

10 Home 22H2

#### Does the issue still occur after restarting your computer?

Yes

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes
